flytekit/tools/translator.py

```diff
@@ -176,6 +177,15 @@ def get_serializable_task(
     )
 
     if isinstance(entity, PythonFunctionTask) and entity.execution_mode == PythonFunctionTask.ExecutionBehavior.DYNAMIC:
+        for e in context_manager.FlyteEntities.entities:
+            if isinstance(e, PythonAutoContainerTask):
+                # 1. Build the ImageSpec for the entities that are inside the dynamic task,
```

Comment:
How does that work? how do we know the entities inside an dynamic task at serialization time?

Reply:
actually, it builds all the entities in the [context](https://github.com/flyteorg/flytekit/blob/9703959508706ca6b3f39e665b3e38fd724978ff/flytekit/core/context_manager.py#L950), not only in dynamic

Reply:
In theory a dynamic wf may reference tasks that don't exist in the static context... We will say you will have to statically declare the task or have it use the same image as the dynamic task. Right?